plugin/clone/src/clone_os.cc

```diff
@@ -289,7 +303,21 @@ int clone_os_copy_buf_to_file(uchar *from_buffer, Ha_clone_file to_file,
     }
 
     auto actual_size = static_cast<uint>(ret_size);
-
+    assert(length + block_pad_len == actual_size);
+
+    if (actual_size > length) {
+      assert(block_pad_len > 0);
+      assert(to_file.o_direct_uneven_file_size);
+      const auto file_size = my_tell(to_file.file_desc, MYF(MY_WME));
+      if (file_size < 0) {
+        return file_size;  // Error code
```

Comment:
I think this should be `return my_errno()`? I think my_tell returns MY_FILEPOS_ERROR (-1).

Reply:
Yeah, `MY_FILEPOS_ERROR` is incorrect. Cannot return `my_errno()` neither, as the function returns errors from the `ER_` space. I am changing it to return `MY_ERROR_ON_WRITE`, like the rest of the function does, because there is no specific `ER_` error for `my_tell`.